### ISSUE TITLE
N813 seems too broad in scope

### ISSUE THREAD

**Issue by Armavica** [astral-sh/ruff#9237]
Hi, currently N813 is raised even for cases such as
``` python
import netCDF4 as nc
```
which seem innocuous from the point of view of the rule: https://docs.astral.sh/ruff/rules/camelcase-imported-as-lowercase/

Would it make sense to restrict the applicability of this rule to cases where `name.to_lowercase() == asname.to_lowercase()`?

**Comment by charliermarsh** [comment]
I'm a bit torn as this does match "upstream" (i.e. ,`pep8-naming` does flag with the same error), but I agree that this isn't a very helpful violation in practice.

**Comment by trag1c** [comment]
> this does match "upstream"

For me that means `pep8-naming` also implemented this in a suboptimal way—this violation shouldn't show up for module aliases imo.
